gulp-tasks/jekyll.js

```diff
@@ -1,11 +1,28 @@
 'use strict';
 
+const fse = require('fs-extra');
 const gulp = require('gulp');
 const path = require('path');
 const spawn = require('child_process').spawn;
 
+
+const addNpmDependencies = () => {
```

Comment:
Could you move this out into a seperate gulp task and have it run before jekyll build / server??